### PR TITLE
Feauture not found page

### PR DESCRIPTION
**Pull request by Mihail18071983**
1Created UI NotFountPage.
2. Added Header components displaying depending on route.

### CODE REVIEW

File: src/app/i18n/locales/all-locales.d.ts [hunk 1]
```diff
@@ -228,6 +228,8 @@ interface AllLocalesType {
     delivery: string[];
     btn_to_basket: string;
     empty_catalog: string[];
+    notFoundContent: string[];
+    goToTheMainPage: string;
```
Comment (chekl) [inline src/app/i18n/locales/all-locales.d.ts]: Please use the case_with_undescore (snake case) style for locals.

File: src/pages/NotFound/index.tsx [hunk 2]
```diff
@@ -1,7 +1,26 @@
...
+import GoToTheMainPageBtn from "shared/UI/Buttons/GoToTheMainPageButton";
+
 export default function NotFound(): JSX.Element {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
   return (
-    <div>
-      <h2>Not found</h2>
-    </div>
+    <section className="px-16 py-[95px]">
```
Comment (chekl) [inline src/pages/NotFound/index.tsx]: Prefer to use tailwind classes. py-24 (96px) https://tailwindcss.com/docs/padding

File: src/shared/UI/Buttons/GoToTheMainPageButton/index.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { useTranslation } from "react-i18next";
+import "../../../../app/styles/index.scss";
```
Comment (chekl) [inline src/shared/UI/Buttons/GoToTheMainPageButton/index.tsx]: You don`t need to import styles in every component (just reminder)
Delete this gotothemainpagebutton component

File: src/pages/NotFound/index.tsx [hunk 4]
```diff
@@ -1,7 +1,26 @@
...
+      <h2 className="text-purple-900 text-[100px] lg:text-[250px]  text-center font-medium leading-[120px] lg:leading-[300px] mb-4 lg:mb-2">
+        404
+      </h2>
+      {Array.from({ length: 2 }).map((_, idx) => (
+        <p className="text-center even:h6 even:lg:h5 even:font-semibold mb-7 odd:additional odd:font-light odd:text-gray-900 odd:lg:h6" key={idx}>
+          {t("notFoundContent")[idx]}
+        </p>
+      ))}
+
+      <GoToTheMainPageBtn
```
Comment (chekl) [inline src/pages/NotFound/index.tsx]: You don`t need to create GoToMainPageBtn component. 
Just use button type="button" className="btn btn-effect {another styles}"

File: src/pages/NotFound/index.tsx [hunk 5]
```diff
@@ -1,7 +1,26 @@
+import { useTranslation } from "react-i18next";
+import { useNavigate } from "react-router-dom";
+import "../../app/styles/index.scss";
```
Comment (chekl) [inline src/pages/NotFound/index.tsx]: delete style import

**Review by chekl — CHANGES_REQUESTED** [review]

File: src/app/i18n/locales/all-locales.d.ts (hunk 1 above)
Comment (Mihail18071983) [reply src/app/i18n/locales/all-locales.d.ts]: Done

File: src/pages/NotFound/index.tsx (hunk 2 above)
Comment (Mihail18071983) [reply src/pages/NotFound/index.tsx]: done

File: src/shared/UI/Buttons/GoToTheMainPageButton/index.tsx (hunk 3 above)
Comment (Mihail18071983) [reply src/shared/UI/Buttons/GoToTheMainPageButton/index.tsx]: done

File: src/pages/NotFound/index.tsx (hunk 4 above)
Comment (Mihail18071983) [reply src/pages/NotFound/index.tsx]: done

File: src/pages/NotFound/index.tsx (hunk 5 above)
Comment (Mihail18071983) [reply src/pages/NotFound/index.tsx]: done

**Pull request merged**